src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -1219,9 +1219,34 @@ public async Task<(long ClusterEtag, string ClusterId)> GenerateClusterIdentityA
             {
                 [id] = identity
             };
-            (var index, var _) = await SendToLeaderAsync(new UpdateClusterIdentityCommand(databaseName, identities));
 
-            return index;
+            var (_, result) = await SendToLeaderAsync(new UpdateClusterIdentityCommand(databaseName, identities));
+
+            if (result == null)
+            {
+                throw new InvalidOperationException(
+                    $"Expected to get result from raft command that should update a cluster-wide identity, but didn't. Leader is {LeaderTag}, Current node tag is {NodeTag}.");
+            }
+
+            var resultDict = result as Dictionary<string, long> ?? throw new InvalidOperationException(
+                                 $"Expected to get result from raft command that should update a cluster-wide identity, but got invalid result structure for {id}. Leader is {LeaderTag}, Current node tag is {NodeTag}.");
+
+            var fullId = $"db/{databaseName}/Identities/{id}".ToLowerInvariant();
+            if (resultDict.ContainsKey(fullId) == false)
```

Comment:
Replace all of this with `TryGetValue`
And why all the complexity, I sent one value it, I'm getting one  value out.

Reply:
The reason for that is UpdateClusterIdentityCommand(databaseName, identities), where `identities` is a dictionary which is used in other places for many values in.  The result, therefor, is many values out.  However at `UpdateClusterIdentityAsync` we do use a single value in/out.  But it will be wrong to return single value from `UpdateClusterIdentityCommand::Execute`